### PR TITLE
feat(integration): DEMO- NCERT-Tutor Agent Integration

### PR DESCRIPTION
**Pull request by internettrashh**
# Introducing NCERT Tutor: A Revolutionary Approach to Simplifying Learning

### Overview

We are excited to propose the addition of NCERT Tutor, a comprehensive learning solution designed to transform the educational landscape. Leveraging advanced technology, NCERT Tutor aims to make learning more accessible, efficient, and engaging for students. This pull request outlines the vision, future scope, and core functionalities of NCERT Tutor, inviting collaboration and feedback from the development team.

### Vision

NCERT Tutor envisions a world where learning is simplified, and students can easily grasp complex concepts through personalized tutoring experiences. Our project targets to:
- Enhance learning attitudes and aptitude among students.
- Provide extensive support to students preparing for exams.
- Automate and ease the processes of memorization and concept revision through autonomous agents.

### Future Scope

We are committed to continuously evolving NCERT Tutor by:
- Introducing a conversational interface to allow students to ask follow-up questions, incorporating their queries into the learning material.
- Implementing interactive flashcards for efficient concept retention and revision, creating an engaging user experience.

#### NCERT Tutor in Action

NCERT Tutor automates the learning process by:
- Fetching relevant chapter content from the NCERT website based on user input.
- Summarizing the content and creating comprehensive notes and question banks.
- Providing a seamless interaction for a personalized tutoring experience.

#### Technology Stack

NCERT Tutor is built with Python, utilizes Fetch.ai for decentralized agent communication, and incorporates various other technologies for a robust and scalable solution.

### Getting Started

Instructions for setting up NCERT Tutor are provided, ensuring an easy start for contributors and users alike.

### Your Feedback and Contributions

We invite the development team to review this pull request. Your feedback and contributions are crucial for the growth and improvement of NCERT Tutor. Together, we can make a significant impact on the educational journey of countless students.

For a closer look at our project and its potential, please refer to the attached presentation and video demo. Let's collaborate to make learning a delightful experience for students worldwide.

### Links for Further Exploration
- deltav  demo [here](https://github.com/fetchai/uAgents/pull/342)

- Video Demo: [Watch Here](https://drive.google.com/drive/folders/175eGJjyQfFjJ36Qk0ACPx_CXWFQbdVnE?usp=drive_link)
- Project Presentation: [View Presentation](https://drive.google.com/file/d/1WZ_CiAbCPbJSSWeq6VmxXqZURk6ZH0Xh/view?usp=sharing)

We look forward to your valuable insights and suggestions. Together, let's shape the future of education.

Thank you for considering NCERT Tutor as an addition to your platform.

### CODE REVIEW

**Review by gautamgambhir97 — CHANGES_REQUESTED** [review]
Sadly this PR does not meet our integrations PR project requirements please update and see https://github.com/fetchai/uAgents/blob/main/integrations/CONTRIBUTING.md, thanks!

**Review by gautamgambhir97 — CHANGES_REQUESTED** [review]

File: integrations/NCERT-Tutor/src/agents/openai/cloud.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import os
+
+
+API_KEY = "813453114559759"
+API_SECRET = "VzDu8Be5e6dtDOvxd8EntitA4iM"
```
[the remark this answers is not in the document]
Comment (internettrashh) [reply integrations/NCERT-Tutor/src/agents/openai/cloud.py]: fixed please check

File: integrations/NCERT-Tutor/src/agents/conversation/question.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+# Import required libraries
+import requests
+from uagents import Model, Protocol, Agent, Context
+from ai_engine import UAgentResponse, UAgentResponseType
+import json
+from ncert import ncert
+from uagents.setup import fund_agent_if_low
+
+AGENT_MAILBOX_KEY = "27c18025-e678-4c1e-840a-cff1d6e969d1"
```
[the remark this answers is not in the document]
Comment (internettrashh) [reply integrations/NCERT-Tutor/src/agents/conversation/question.py]: fixed please check

**Comment by abhifetch** [comment]
This is a hackathon example and hence needs to be moved from integrations to hackathon examples. Closing the PR in integrations and creating a new PR in hackathon examples with a reference to this one.

**Pull request closed without merging**